platform/core/src/services/ToolBarService/ToolbarService.ts

```diff
@@ -1,126 +1,169 @@
-import merge from 'lodash.merge';
 import { CommandsManager } from '../../classes';
 import { ExtensionManager } from '../../extensions';
 import { PubSubService } from '../_shared/pubSubServiceInterface';
-import type { RunCommand, Commands } from '../../types/Command';
+import type { RunCommand } from '../../types/Command';
+import ServicesManager from '../ServicesManager';
 
 const EVENTS = {
   TOOL_BAR_MODIFIED: 'event::toolBarService:toolBarModified',
   TOOL_BAR_STATE_MODIFIED: 'event::toolBarService:toolBarStateModified',
 };
 
-export type ButtonListeners = Record<string, RunCommand>;
+export type EvaluatePublic = string | EvaluateFunction;
 
-export interface ButtonProps {
-  primary?: Button;
-  secondary?: Button;
-  items?: Button[];
-}
+export type EvaluateFunction = (props: Record<string, unknown>) => {
+  disabled: boolean;
+  className: string;
+};
 
-export interface Button extends Commands {
+export interface ButtonProps {
   id: string;
-  icon?: string;
-  label?: string;
-  type?: string;
+  icon: string;
+  label: string;
   tooltip?: string;
-  isActive?: boolean;
-  listeners?: ButtonListeners;
-  props?: ButtonProps;
+  commands?: RunCommand;
+  disabled?: boolean;
+  className?: string;
+  evaluate?: EvaluatePublic;
+  listeners?: Record<string, RunCommand>;
+}
+
+export interface NestedButtonProps {
```

Comment:
Should be a type here - these are simple typed objects, not interfaces that we expect to be implemented AFAIK.